compiler/generator_jsx.c

```diff
@@ -0,0 +1,1495 @@
+ * current string */
+static void generate_dollar(struct generator * g, struct node * p) {
+
+    struct str * savevar = vars_newname(g);
+    write_comment(g, p);
+    g->V[0] = p->name;
+
+    str_assign(g->failure_str, "this.copy_from(");
+    str_append(g->failure_str, savevar);
+    str_append_string(g->failure_str, ");");
+    g->B[0] = str_data(savevar);
+    writef(g, "~{~M~n ~B0 = this;~N"
+             "~Mthis.current = this.~V0.toString();~N"
+             "~Mthis.cursor = 0;~N"
+             "~Mthis.limit = (this.current.length);~N", p);
+    generate(g, p->left);
+    if (!g->unreachable) {
+        write_margin(g);
+        write_str(g, g->failure_str);
+        write_newline(g);
+    }
+    w(g, "~}");
+    str_delete(savevar);
+}
+
+static void generate_integer_assign(struct generator * g, struct node * p, char * s) {
+
+    g->V[0] = p->name;
+    g->S[0] = s;
+    if (p->AE->type == c_name)
+    {
+        w(g, "~Mthis.~V0 ~S0 this."); generate_AE(g, p->AE); w(g, ";~N");
+    }
+    else
+    {
+        w(g, "~Mthis.~V0 ~S0 "); generate_AE(g, p->AE); w(g, ";~N");
+    }
+}
+
+static void generate_integer_test(struct generator * g, struct node * p, char * s) {
+
+    g->V[0] = p->name;
+    g->S[0] = s;
+    if (p->AE->type == c_name)
+    {
+        w(g, "~Mif (!(this.~V0 ~S0 this."); generate_AE(g, p->AE); w(g, "))~N");
+    }
+    else
+    {
+        w(g, "~Mif (!(this.~V0 ~S0 ");
+        generate_AE(g, p->AE);
+        w(g, "))~N");
+    }
+    write_block_start(g);
+    write_failure(g);
+    write_block_end(g);
+    g->unreachable = false;
+}
+
+static void generate_call(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->V[0] = p->name;
+    write_failure_if(g, "!this.~V0()", p);
+}
+
+static void generate_grouping(struct generator * g, struct node * p, int complement) {
+
+    struct grouping * q = p->name->grouping;
+    g->S[0] = p->mode == m_forward ? "" : "_b";
+    g->S[1] = complement ? "out" : "in";
+    g->V[0] = p->name;
+    g->I[0] = q->smallest_ch;
+    g->I[1] = q->largest_ch;
+    if (q->no_gaps)
+        write_failure_if(g, "!(this.~S1_range~S0(~I0, ~I1))", p);
+    else
+        write_failure_if(g, "!(this.~S1_grouping~S0(~n.~V0, ~I0, ~I1))", p);
+}
+
+static void generate_namedstring(struct generator * g, struct node * p) {
+
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "" : "_b";
+    g->V[0] = p->name;
+    write_failure_if(g, "!(this.eq_v~S0(this.~V0))", p);
+}
+
+static void generate_literalstring(struct generator * g, struct node * p) {
+
+    symbol * b = p->literalstring;
+    write_comment(g, p);
+    g->S[0] = p->mode == m_forward ? "" : "_b";
+    g->I[0] = SIZE(b);
+    g->L[0] = b;
+    write_failure_if(g, "!(this.eq_s~S0(~I0, ~L0))", p);
+}
+
+static void generate_define(struct generator * g, struct node * p) {
+
+    struct name * q = p->name;
+    symbol stem[] = {'s', 't', 'e', 'm'};
```

Comment:
This seems to be special casing functions named `stem`, which is clearly not right.  It seems to mark them as `override`.  Not entirely sure what this is all for, but probably it should instead check if the function is `external`.